src/components/Button/Button.types.ts

```diff
@@ -44,15 +55,36 @@ export interface SplitButtonProps
         'text' | 'htmlType' | 'onContextMenu' | 'splitButtonProps' | 'toggle'
     > {}
 
-export interface InternalButtonProps extends ButtonProps {
+export interface TwoStateButtonProps
+    extends Omit<
+        InternalButtonProps,
+        | 'htmlType'
+        | 'icon'
+        | 'iconColor'
+        | 'onContextMenu'
+        | 'split'
+        | 'splitButtonProps'
+    > {
     /**
-     * Determines the button type.
+     * The button counter number.
      */
-    type?: ButtonType;
+    counter?: number;
     /**
-     * Ref of the button
+     * The button icon 1.
      */
-    ref?: Ref<HTMLButtonElement>;
+    iconOne?: IconName;
```

Comment:
We also have Icon only two state buttons, for eg starring candidate. Will this component handle that, or that will be its own thing ?